packages/jellyfish-api-core/__tests__/category/icxorderbook/submitExtHTLC.test.ts

```diff
@@ -0,0 +1,524 @@
+
+    const accountDFIBeforeExtHTLC: Record<string, BigNumber> = await client.call('getaccount', [accountDFI, {}, true], 'bignumber')
+    // submit EXT HTLC - maker
+    const ExtHTLC: ExtHTLC = {
+      offerTx: makeOfferTxId,
+      amount: new BigNumber(1),
+      hash: '957fc0fd643f605b2938e0631a61529fd70bd35b2162a21d978c41e5241a5220',
+      htlcScriptAddress: '13sJQ9wBWh8ssihHUgAaCmNWJbBAG5Hr9N',
+      ownerPubkey: '036494e7c9467c8c7ff3bf29e841907fb0fa24241866569944ea422479ec0e6252',
+      timeout: 100
+    }
+    const ExtHTLCTxId = (await client.icxorderbook.submitExtHTLC(ExtHTLC)).txid
+    await container.generate(1)
+
+    // List htlc
+    const listHTLCOptions = {
+      offerTx: makeOfferTxId
+    }
+    const HTLCs: Record<string, ICXDFCHTLCInfo | ICXEXTHTLCInfo | ICXClaimDFCHTLCInfo> = await client.call('icx_listhtlcs', [listHTLCOptions], 'bignumber')
+    expect(Object.keys(HTLCs).length).toBe(2) // extra entry for the warning text returned by the RPC atm.
+    expect(HTLCs[ExtHTLCTxId] as ICXEXTHTLCInfo).toStrictEqual(
+      {
+        type: ICXHTLCType.EXTERNAL,
+        status: ICXOrderStatus.OPEN,
+        offerTx: makeOfferTxId,
+        amount: ExtHTLC.amount,
+        amountInDFCAsset: ExtHTLC.amount.multipliedBy(order.orderPrice),
+        hash: ExtHTLC.hash,
+        htlcScriptAddress: ExtHTLC.htlcScriptAddress,
+        ownerPubkey: ExtHTLC.ownerPubkey,
+        timeout: new BigNumber(ExtHTLC.timeout),
+        height: expect.any(BigNumber)
+      }
+    )
+
+    const accountDFIAfterEXTHTLC: Record<string, BigNumber> = await client.call('getaccount', [accountDFI, {}, true], 'bignumber')
+    // maker deposit should be reduced from accountDFI
+    expect(accountDFIAfterEXTHTLC[idDFI]).toStrictEqual(accountDFIBeforeExtHTLC[idDFI].minus(0.1))
+  })
+
+  it('should submit ExtHTLC for a DFC buy offer with input utxos', async () => {
+    const { order, createOrderTxId } = await icxSetup.createDFISellOrder('BTC', accountDFI, '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941', new BigNumber(15), new BigNumber(0.01))
+    const { makeOfferTxId } = await icxSetup.createDFIBuyOffer(createOrderTxId, new BigNumber(0.10), accountBTC)
+    await icxSetup.createDFCHTLCForDFIBuyOffer(makeOfferTxId, new BigNumber(10), '957fc0fd643f605b2938e0631a61529fd70bd35b2162a21d978c41e5241a5220', 500)
+
+    const accountBTCBeforeEXTHTLC: Record<string, BigNumber> = await client.call('getaccount', [accountBTC, {}, true], 'bignumber')
+    // submit EXT HTLC - taker
+    const ExtHTLC: ExtHTLC = {
+      offerTx: makeOfferTxId,
+      amount: new BigNumber(0.10),
+      hash: '957fc0fd643f605b2938e0631a61529fd70bd35b2162a21d978c41e5241a5220',
+      htlcScriptAddress: '13sJQ9wBWh8ssihHUgAaCmNWJbBAG5Hr9N',
+      ownerPubkey: '036494e7c9467c8c7ff3bf29e841907fb0fa24241866569944ea422479ec0e6252',
+      timeout: 15
+    }
+
+    // input utxos
+    const inputUTXOs = await container.fundAddress(accountBTC, 10)
```

Comment:
Can you change the name to `utxos`

Reply:
Like how you declare in your other functions. For eg:

<img width="814" alt="Screenshot 2021-07-07 at 12 54 23 PM" src="https://user-images.githubusercontent.com/7101806/124702155-8351f280-df22-11eb-87e7-3d33b8e47dc8.png">

Reply:
should not be a problem since this is a variable name inside a test?

Reply:
Okay sure.